packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -2,9 +2,1489 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
 import 'package:meta/meta.dart';
 
-import 'print.dart';
+/// Styles for displaying a node in a [DiagnosticsNode] tree.
+///
+/// See also:
+///
+///  * [DiagnosticsNode.toStringDeep], which dumps text art trees for these
+///    styles.
+enum DiagnosticsTreeStyle {
+  /// Sparse style for displaying trees.
+  ///
+  /// See also:
+  ///
+  ///  * [RenderObject], which uses this style.
+  sparse,
+
+  /// Style that indicates a node is offstage.
+  offstage,
+
+  /// Slightly more compact version of the [sparse] style.
+  ///
+  /// See also:
+  ///
+  ///  * [Element], which uses this style.
+  dense,
+
+  /// Surround a node in the tree with a box.
+  ///
+  /// See also:
+  ///
+  ///  * [RenderParagraph], which uses this style.
+  box,
+
+  /// Render the tree just using whitespace without connecting parents to
+  /// children using lines.
+  ///
+  /// See also:
+  ///
+  ///  * [SliverGeometry], which uses this style.
+  whitespace,
+
+  /// Render the tree on a single line without showing children.
+  singleLine,
+}
+
+/// Configuration specifying how a particular [DiagnosticsTreeStyle] should be
+/// rendered as text art.
+///
+/// The configuration is generally applied as follows to dump a
+/// [DiagnosticsNode] as text art.
+///
+/// {diagnosticsNode.name} [afterName]
+/// {diagnosticsNode.description} [afterDescriptionIfBody] [beforeProperties] [lineBreak]
+/// [linkCharacter] {property1} [propertySeparator] [lineBreak]
+/// [linkCharacter] {property2} [propertySeparator] [lineBreak]
+/// [linkCharacter] ...
+/// [linkCharacter] {propertyN} [afterProperties] [lineBreak]
+/// [prefixLineOne] {child1} [lineBreak]
+/// [linkCharacter] [prefixOtherLines] [vertical horizontalSpace {child1 content} [lineBreak]
+/// [prefixLineOne] horizontal {child2 line one } [lineBreak]
+/// [prefixOtherLines] {child2 other lines} [lineBreak]
+/// ...
+/// [prefixLineOneLastChild ] {childN} [lineBreak]
+/// [childLinkSpace] {childN other lines} [lineBreak]
+/// [footer]
```

Comment:
This looks confusing. I would either omit it or create a diagram, like this: https://master-docs-flutter-io.firebaseapp.com/flutter/material/AppBar-class.html (as created by https://github.com/flutter/assets-for-api-docs/blob/master/material/app_bar.dart).